src/Packetery/Module/Order/CarrierModal.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Class CarrierModal.
+ *
+ * @package Packetery
+ */
+
+declare( strict_types=1 );
+
+namespace Packetery\Module\Order;
+
+use Packetery\Module;
+use Packetery\Latte\Engine;
+use Packetery\Nette\Forms;
+
+/**
+ * Class CarrierModal.
+ *
+ * @package Packetery
+ */
+class CarrierModal {
+
+	public const MODAL_ID = 'wc-packetery-carrier-modal';
+
+	/**
+	 * Latte engine.
+	 *
+	 * @var Engine
+	 */
+	private $latteEngine;
+
+	/**
+	 * Context resolver.
+	 *
+	 * @var Module\ContextResolver
+	 */
+	private $contextResolver;
+
+	/**
+	 * Order detail common logic.
+	 *
+	 * @var DetailCommonLogic
+	 */
+	private $detailCommonLogic;
+
+	/**
+	 * Order repository.
+	 *
+	 * @var CarrierModalFormFactory
+	 */
+	private $carrierModalFormFactory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Engine                  $latteEngine             Latte engine.
+	 * @param DetailCommonLogic       $detailCommonLogic       Detail common logic.
+	 * @param Module\ContextResolver  $contextResolver         Context resolver.
+	 * @param CarrierModalFormFactory $carrierModalFormFactory Carrier Modal form factory.
+	 */
+	public function __construct(
+		Engine $latteEngine,
+		DetailCommonLogic $detailCommonLogic,
+		Module\ContextResolver $contextResolver,
+		CarrierModalFormFactory $carrierModalFormFactory
+	) {
+		$this->latteEngine             = $latteEngine;
+		$this->contextResolver         = $contextResolver;
+		$this->detailCommonLogic       = $detailCommonLogic;
+		$this->carrierModalFormFactory = $carrierModalFormFactory;
+	}
+
+	/**
+	 * Registers order modal.
+	 *
+	 * @return void
+	 */
+	public function register(): void {
+		add_action( 'admin_head', [ $this, 'renderCarrierModal' ] );
+	}
+
+	/**
+	 * Renders modal.
+	 *
+	 * @return void
+	 */
+	public function renderCarrierModal(): void {
+		if ( false === $this->contextResolver->isOrderDetailPage()
+			&& false === $this->detailCommonLogic->isPacketeryOrder()
```

Comment:
Stačí isPacketeryOrder - podmínku na isOrderDetailPage() v sobě již obsahuje.
Podmínka je navíc špatně sestavená, co se stane, pokud je to detail, ale není to naše objednávka?

Reply:
Rozumim, opraveno.